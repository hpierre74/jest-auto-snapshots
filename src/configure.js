import { createElement } from 'react';

const prepend = 'jest-auto-snapshots';
export const NodeFixture = () => createElement('span', {}, `${prepend} Node Fixture`);
const stringFixture = `${prepend} String Fixture`;

let config = {
  fixtures: {
    string: stringFixture,
    bool: true,
    number: 1,
    node: createElement(NodeFixture),
    element: createElement(NodeFixture),
    func: jest.fn(),
    any: stringFixture,
    symbol: Symbol(stringFixture),
    array: [stringFixture],
    object: { jest: stringFixture },
  },
  relativePath: '../',
  extension: 'jsx',
};

export const set = (rootOptions) => {
  config = {
    ...config,
    ...rootOptions,
    fixtures: {
      ...config.fixtures,
      ...(rootOptions.fixtures || {}),
    },
  };
};

export const get = () => config;
