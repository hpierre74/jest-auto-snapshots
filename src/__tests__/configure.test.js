import React from 'react';
import { shallow } from 'enzyme';
import { set, get, NodeFixture } from '../configure.js';

const defaultConfig = get();

describe('configure', () => {
  describe('NodeFixture', () => {
    it('Should match snapshot', () => {
      expect(shallow(<NodeFixture />)).toMatchSnapshot();
    });
  });

  describe('set', () => {
    it('Should set relativePath and extension correctly w/o fixtures', () => {
      set({
        relativePath: 'mock',
        extension: 'js',
      });
      expect(get()).toEqual({
        ...defaultConfig,
        relativePath: 'mock',
        extension: 'js',
      });
    });

    it('Should set fixtures correctly', () => {
      const config = get();
      const newFixtures = {
        mockCustomPropType: 'mock-new-fixture',
        string: 'mock-override-default-fixture',
      };
      set({
        fixtures: newFixtures,
      });
      expect(get()).toEqual({
        ...config,
        fixtures: {
          ...config.fixtures,
          ...newFixtures,
        },
      });
    });
  });
});
