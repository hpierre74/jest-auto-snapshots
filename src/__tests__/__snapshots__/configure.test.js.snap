// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`configure NodeFixture Should match snapshot 1`] = `
<span>
  jest-auto-snapshots Node Fixture
</span>
`;
